packages/graph/hash_graph/lib/graph/src/ontology/data_type.rs

```diff
@@ -35,6 +35,58 @@ impl QueryRecord for DataType {
     type Path<'q> = DataTypeQueryPath;
 }
 
+impl RecordPath for DataTypeQueryPath {
+    fn expected_type(&self) -> ParameterField {
+        match self {
+            Self::OwnedById | Self::CreatedById | Self::UpdatedById => ParameterField {
+                parameter_type: ParameterType::Uuid,
+                optional: false,
+            },
+            Self::RemovedById => ParameterField {
+                parameter_type: ParameterType::Uuid,
+                optional: true,
+            },
+            Self::BaseUri => ParameterField {
+                parameter_type: ParameterType::BaseUri,
+                optional: false,
+            },
+            Self::VersionedUri => ParameterField {
+                parameter_type: ParameterType::VersionedUri,
+                optional: false,
+            },
+            Self::Version => ParameterField {
+                parameter_type: ParameterType::UnsignedInteger,
+                optional: false,
+            },
+            Self::Description => ParameterField {
+                parameter_type: ParameterType::Text,
+                optional: true,
```

Comment:
I'm not familiar enough with this design yet so just consider this a guiding question. 

Do we need to store this information? 

It feels like it's another place for us to let the code go out of sync in nasty breaking ways with edge-cases. What if we change the schema definition to make something optional, or make it not optional, but don't remember to update these?

Reply:
I hear your concern hear. I don't really know if we want to allow checking for `NULL` on non-optional type. This could potentially avoid bugs in downstream code at "compile time". But this is only easily possible for well-defined types (like ontology types), but will be hard to check for properties. I don't mind removing the `optional` field, what do you think?

Reply:
I don't quite understand. How would the optional thing be used? 

Your "compile time" comment makes me think it's trying to stop users from creating invalid structural queries. If that's the case, what's an example of an invalid structural query in this?

Reply:
Currently, it's not used.

Yes, exactly. An invalid query could be `version == NULL`

Reply:
I think we don't need to account for that, I'd just remove them for now as they're complicating things

Reply:
Removed it in 61557e89e9af0b0205a781d84e90d96a2072f962